test/e2e/metrics_test.go

```diff
@@ -155,6 +155,10 @@ func getClientWithCertAuth(t *testing.T) *http.Client {
 	// that are missing from the system trust roots
 	rootCAs.AppendCertsFromPEM(config.CAData)
 
+	// TODO: could get router-ca ConfigMap we sync'd from openshift-config-managed
```

Comment:
Not really, you're using a passthrough route to access the /metrics endpoint from an external endpoint, but the cert that's used for serving directly at the pod which terminates the TLS connection is not valid for the hostname of the route (and is actually signed by service-ca AFAIK).

So even if you had a valid CA that signed the server cert here, the cert verification would fail because of the hostname. Yet you need passthrough route to be able to use client cert auth. Another solution would be to grab a token for a service account that's allowed to access this endpoint, in which case the passthrough aspect of the route would not be necessary.